main.go

```diff
@@ -179,6 +180,29 @@ func main() {
 		os.Exit(1)
 	}
 
+	decoder := admission.NewDecoder(mgr.GetScheme())
+
+	if os.Getenv("DISABLE_AUTO_ANNOTATION") != "true" {
```

Comment:
nit: Can we type cast these env vars as boolean, so that we do not check them against string versions of bools?